corporate/lib/stripe.py

```diff
@@ -1166,6 +1169,31 @@ def update_billing_modality_of_current_plan(self, charge_automatically: bool) ->
             success_message = f"Billing collection method of {self.billing_entity_display_name} updated to send invoice."
         return success_message
 
+    def update_end_date_of_current_plan(self, end_date_string: str) -> str:
+        new_end_date = datetime.strptime(end_date_string, "%Y-%m-%d").replace(tzinfo=timezone.utc)
+        if new_end_date.date() <= timezone_now().date():
```

Comment:
Since the input is a date (without a time), it seemed best for the check here to look at the current UTC date and the date from the form input. But perhaps just comparing the two datetime values would be fine.

In general, this support action should be about extending a plan to be longer. The modify plan actions for downgrading a plan would be ideally used for ending a plan sooner rather than later.